components/tidb_query/src/codec/chunk/chunk.rs

```diff
@@ -63,75 +63,110 @@ impl Chunk {
     #[inline]
     pub fn append_vec(
         &mut self,
-        row_index: usize,
+        row_indexs: &[usize],
         field_type: &FieldType,
         vec: &VectorValue,
         column_index: usize,
     ) -> Result<()> {
         let col = &mut self.columns[column_index];
         match vec {
-            VectorValue::Int(ref vec) => match &vec[row_index] {
-                None => {
-                    col.append_null().unwrap();
-                }
-                Some(val) => {
-                    if field_type
-                        .as_accessor()
-                        .flag()
-                        .contains(FieldTypeFlag::UNSIGNED)
-                    {
-                        col.append_u64(*val as u64).unwrap();
-                    } else {
-                        col.append_i64(*val).unwrap();
+            VectorValue::Int(ref vec) => {
+                for row_index in row_indexs {
+                    let row_index = *row_index as usize;
+                    match &vec[row_index] {
+                        None => {
+                            col.append_null().unwrap();
+                        }
+                        Some(val) => {
+                            if field_type
+                                .as_accessor()
+                                .flag()
+                                .contains(FieldTypeFlag::UNSIGNED)
```

Comment:
Small nit: we can avoid the condition calculation and the branch inside the loop